sdk/core/System.Net.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net.ClientModel.Core;
+using System.Net.ClientModel.Internal;
+
+namespace System.Net.ClientModel
+{
+    /// <summary>
+    /// Provides functionality to read and write <see cref="IModel{T}"/> and <see cref="IJsonModel{T}"/>.
+    /// </summary>
+    public static class ModelReaderWriter
```

Comment:
Another non-PR-blocking question/comment, more as questions to prep for BCL review...  Is there a need for this type to do both reading and writing?  What is it?  I'm wondering, given the at-first-glance niceness of interface names like `IModelReadable` and `IModelWriteable` how disruptive/costly to the implementation it would be to split components for reading and writing to maybe simplify the APIs?  I feel like we often get developer and partner questions like "why didn't my model get a serialize method generated?"  For an output-only model, we don't tend to generate the serialize methods, so will coupling read and write mean that we are changing team precedent regarding that aspect of today's model design?  If we split them, output-only models could feasibly only implement the `IModelReadable` interface, and it might be more clear what is happening there and why...

Reply:
We will always create both the Write method and Read method for all models going forward regardless of usage.  The reason for this is the use case for a customer is to be able to store the full state of the model.  The fact that the model might be output only or input only doesn't matter they will need both Read and Write.